File: src/main/java/org/example/chessplatformbe/controller/dto/request/CreateCommentDTO.java
package org.example.chessplatformbe.controller.dto.request;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CreateCommentDTO {

    private Integer updateId;

    @NotBlank(message = "Text is required")
    private String text;

    @NotNull(message = "User id is required")
    private Integer userId;

    @NotNull(message = "Article id is required")
    private Integer articleId;
}

File: src/controller/dto/request/create_comment_dto.rs
pub struct CreateCommentDTO {
    update_id: u32,
    text: String,
    user_id: u32,
    article_id: u32,
}

impl CreateCommentDTO {
    pub fn new(update_id: u32, text: String, user_id: u32, article_id: u32) -> Self {
        Self {
            update_id,
            text,
            user_id,
            article_id,
        }
    }
}
